generator/namespaces.py

```diff
@@ -0,0 +1,88 @@
+"""Generate namespaces.yaml."""
+import tarfile
+import urllib
+import warnings
+from io import BytesIO
+from itertools import groupby
+from operator import itemgetter
+from pathlib import Path
+
+import click
+import requests
+import yaml
+
+
+def _get_first(tuple_):
+    return tuple_[0]
+
+
+def _get_views(uri):
+    with urllib.request.urlopen(uri) as f:
+        tarbytes = BytesIO(f.read())
+    views = {}
+    with tarfile.open(fileobj=tarbytes, mode="r:gz") as tar:
+        for tarinfo in tar:
+            if tarinfo.name.endswith("/view.sql"):
+                *_, project, dataset_id, view_id, _ = tarinfo.name.split("/")
+                if project == "moz-fx-data-shared-prod":
+                    if dataset_id not in views:
+                        views[dataset_id] = {}
+                    views[dataset_id][view_id] = tar.extractfile(tarinfo.name)
+    return views
+
+
+@click.command(help=__doc__)
+@click.option(
+    "--custom-namespaces",
+    default=None,
+    type=click.File(),
+    help="Path to a custom namespaces file",
+)
+@click.option(
+    "--generated-sql-uri",
+    default="https://github.com/mozilla/bigquery-etl/archive/generated-sql.tar.gz",
+    help="URI of a tar archive of the bigquery-etl generated-sql branch, which is "
+    "used to list views and determine whether they reference stable tables",
+)
+def namespaces(custom_namespaces, generated_sql_uri):
+    """Generate namespaces.yaml."""
+    warnings.filterwarnings("ignore", module="google.auth._default")
+
+    get_app_name = itemgetter("app_name")
+    app_listings = sorted(
```

Comment:
Does this need to be sorted, since we're using `groupby` later?

Reply:
yes, that is why this is sorted. i'll add a comment